airbyte-cdk/python/airbyte_cdk/sources/declarative/decoders/json_decoder.py

```diff
@@ -17,8 +18,43 @@ class JsonDecoder(Decoder):
 
     parameters: InitVar[Mapping[str, Any]]
 
-    def decode(self, response: requests.Response) -> Union[Mapping[str, Any], List[Any], Any]:
+    def decode(self, response: requests.Response) -> Generator[Mapping[str, Any], None, None]:
         try:
-            return response.json()
+            body_json = response.json()
+            if not isinstance(body_json, list):
+                body_json = [body_json]
+            yield from body_json
         except requests.exceptions.JSONDecodeError:
-            return {}
+            yield {}
+
+
+@dataclass
+class IterableDecoder(Decoder):
```

Comment:
is this decoder useful for any existing connectors?

Reply:
This one is used in iterable
@@ -357,6 +358,8 @@
https://github.com/airbytehq/airbyte/pull/38829#issuecomment-2154399045